### ISSUE TITLE
Markdown image link to Attlasian wiki format wrong conversion in case of page titles with uncommon chars like square braces

### ISSUE THREAD

**Issue by qwazer** [bsorrentino/maven-confluence-plugin#191]
I switched from 6.0-rc4 to 6.3.1 and found the problem with posting *md pages that contains links like

Markdown source
```
![cb-integration-components.png](cb-integration-components.png)
```

Produced Attlasian wiki output in 6.0
```
!cb-integration-components.png|alt="cb-integration-components.png"!
```

Produced Attlasian wiki output in 6.3.1

```
![Integration] CB^cb-integration-components.png|cb-integration-components.png!
```

This cause the problem when use XML/RPC API
```
Caused by: java.lang.Exception: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: [com.ctc.wstx.exc.WstxLazyException] com.ctc.wstx.exc.WstxParsingException: Unexpected '<'  in attribute value
 at [row,col {unknown-source}]: [470,100]
    at org.bsc.confluence.xmlrpc.Confluence.call (Confluence.java:937)
    at org.bsc.confluence.xmlrpc.Confluence.storePage (Confluence.java:324)
    at org.bsc.confluence.xmlrpc.XMLRPCConfluenceServiceImpl.storePage (XMLRPCConfluenceServiceImpl.java:248)
    at org.bsc.confluence.ConfluenceServiceFactory$MixedConfluenceService.storePage (ConfluenceServiceFactory.java:96)
    at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.lambda$updateHomeContent$2 (ConfluenceDeployMojo.java:573)
    at org.bsc.confluence.model.SiteProcessor.processPageUri (SiteProcessor.java:146)
```

Can revert to previous behaviur?

**Comment by bsorrentino** [comment]
Hi @qwazer 

I prefer try to keep as much as possible compatibility between version.

do you think that is not possible to achieve it?

**Comment by qwazer** [comment]
> I prefer try to keep as much as possible compatibility between versions.

Does it mean that you want to keep current behaviour with adding page titles to generated wiki text in 6.X?

> do you think that is not possible to achieve it?

It depends on the Confluence wiki format. If it is possible to use encoding, etc in page titles inside wiki format, then the current behaviour of the plugin can be preserved.

**Comment by qwazer** [comment]
refers to #187 #189

**Comment by bsorrentino** [comment]
hi @qwazer 

thanks for details .. I'll start to review the code asap to find out the best trade off

**Comment by bsorrentino** [comment]
hi @qwazer 

I've updated test  [Issue189Test](https://github.com/bsorrentino/maven-confluence-plugin/blob/develop/maven-confluence-core/src/test/java/org/bsc/maven/plugin/confluence/Issue189Test.java)  in core module to verify your condition and seems that the parser work well also on current version in fact it generate `!${page.title}^cb-integration-components.png|cb-integration-components.png!`

Please could you check and let me know ? 

Thanks in advance

**Comment by qwazer** [comment]
@bsorrentino Yes the test is OK, But the problem begins when page title contains square braces [].
In our company these are widely used in confluence page titles.

When I try t post page with the title `[Integration] CB` and with attachment inside
```
![Integration] CB^cb-integration-components.png|cb-integration-components.png!
``` 
It produces errors above.
If I change title to simple `CB` it works fine.

**Comment by qwazer** [comment]
I investigated it more deeply.

In case of XML-RPC Confluence produce error immediately.
In case of REST API Confluence doesn't produce error, but don't show attachment on the page.


I compared produced Storage Format in Confluence

The actual page title is `[Design review] [Prism] CB integration module`

Previously (plugin version 6.0-rc4):
```xml
<p><ac:image ac:alt="cb-integration-components.png"><ri:attachment ri:filename="cb-integration-components.png" /></ac:image></p>
```

Current with REST API (plugin version 6.3.1): - it produces unknown attachment in page

```xml 
<p><ac:image><ri:attachment ri:filename="cb-integration-components.png"><ri:page ri:content-title="&lt;ac:link&gt;&lt;ri:page ri:content-title=" /> <ac:link><ri:page ri:content-title="Prism" /></ac:link> CB integration module&quot; /&gt;</ri:attachment></ac:image></p>
```

May be to add some condition to not add titles if it was not used in original markdown file? Looks like it will solve the problem.

**Comment by bsorrentino** [comment]
hi @qwazer 

Thanks for great analysis, I'll try to find a solution

**Comment by bsorrentino** [comment]
Hi @qwazer 

I've update markdown processor to make it bit smarter. It means that the processor detects special characters (currently `square bracket`) in the title and switch off adding of `${page.title}` prefix

I've deployed develop version `6.3.2-SNAPSHOT` could you test it and let me know ?

**Comment by qwazer** [comment]
HI @bsorrentino 
The plugin of version 6.3.2-SNAPSHOT work nice for me!
Thanks!

**Comment by bsorrentino** [comment]
Hi @qwazer 

Thanks for quick feedback ... I'll arrange release soon

**Comment by bsorrentino** [comment]
fix in release `6.3.2`
